File: 0268-missing-number/0268-missing-number.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
     
        
        
        //using cyclic sort 
        
        int i=0;
        while(i<nums.size()){
            int correctPos = nums[i];
            if( nums[i] < nums.size() && nums[i] != nums[correctPos]){
                swap(nums[correctPos], nums[i]);
            }
            else{
                i++;
            }
        }
        for(int i=0; i<nums.size(); i++){
            if(nums[i]!=i){
                return i;
            }
        }
        return nums.size();
    }
};

File: 0268-missing-number/0268-missing-number.java
class Solution {
    public int missingNumber(int[] arr) { 
        
     int H[] = new int[arr.length+1];
        Arrays.fill(H,0);
        
        for(int i=0; i<arr.length; i++){
            H[arr[i]]++;
        }
    
        for(int i=1; i<H.length; i++){
            if(H[i]==0){
                return i;
            }
        }
        return 0;
  }
}